### DISCUSSION TITLE
Logging for run_maintenance_proc

### DISCUSSION THREAD

**Discussion by raags** [pgpartman/pg_partman#475] [General]
Is there a way to check what was done during a run? The procedure does not produce any output. I'm using `pg_cron` on AWS RDS to schedule the run, and it would be useful to have information on what was done, and if there were errors if any. 

Maybe the output could be written to a table? 

<I saw there is another discussion on a similar question, but that veered off into something else, so starting this new one>

**Comment by keithf4** [comment]
The pg_jobmon extension is supported to provide logging and monitoring of pg_partman. 

https://github.com/omniti-labs/pg_jobmon

The issue with writing to a table is that if the function/procedure fails, whatever was last inserted into that table as part of the last transaction will also be undone. pg_jobmon solves that by using dblink. This does cause some slight overhead since it has to make new connections for each step logged, but most times this isn't a concern. Please test to be sure though.

You can also monitor your postgresql logs for any errors as well.

**Reply by raags** [reply]
`pg_jobmon` is, unfortunately, isn't available on AWS RDS. 

I see that `NOTICE` is the log level used by pg_partman, just above the default `WARNING`. So I'll set `log_min_messages` to that - it shouldn't generate too many messages.
